addons/account/models/account_move.py

```diff
@@ -1452,6 +1452,8 @@ def _get_last_sequence_domain(self, relaxed=False):
             if self.journal_id.refund_sequence:
                 refund_types = ('out_refund', 'in_refund')
                 domain += [('move_type', 'in' if self.move_type in refund_types else 'not in', refund_types)]
+            if self.journal_id.payment_sequence:
+                domain += [('payment_id', '!=' if self.payment_id else '=', False)]
```

Comment:
The problem here and in other places where payment_id is used, is that it doesn't seem to be set yet.
I need another condition to set apart payments and other moves, help appreciated @smetl

Reply:
Or maybe a way to delay the evaluation / force payment_id to be computed ?